lib/web_ui/lib/src/engine/semantics/heading.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../dom.dart';
+import 'semantics.dart';
+
+/// Renders semantics objects as headings with the corresponding
+/// level (h1 ... h6).
+class Heading extends PrimaryRoleManager {
+  Heading(SemanticsObject semanticsObject)
+      : super.withBasics(PrimaryRole.heading, semanticsObject) {
+        addHeadingRole();
+      }
+
+  static const int defaultHeadingLevel = 1;
+
+  @override
+  void update() {
+    super.update();
```

Comment:
We can return early if the heading level field is not dirty.

Reply:
Ok, I added it.